examples/medplum-patient-intake-demo/src/bots/core/intake-form.tsx

```diff
@@ -0,0 +1,137 @@
+import {
+  BotEvent,
+  createReference,
+  getExtension,
+  getQuestionnaireAnswers,
+  getReferenceString,
+  MedplumClient,
+  resolveId,
+} from '@medplum/core';
+import { Coding, Extension, HumanName, Observation, Patient, QuestionnaireResponse } from '@medplum/fhirtypes';
+
+export async function handler(event: BotEvent<QuestionnaireResponse>, medplum: MedplumClient): Promise<void> {
+  const response = event.input;
+  const answers = getQuestionnaireAnswers(response);
+
+  if (!response.subject) {
+    return;
+  }
+
+  const patient = await medplum.readResource('Patient', resolveId(response.subject) as string);
+
+  if (!patient) {
+    return;
+  }
+
+  // Handle demographic information
+
+  const newName = {
+    given: [answers['first-name'].valueString, answers['middle-name'].valueString],
+    family: answers['last-name'].valueString,
+  } as HumanName;
+
+  patient.name = [newName];
+  patient.birthDate = answers['dob'].valueDate;
+  patient.gender = answers['gender-identity'].valueCoding?.code as Patient['gender'];
+
+  setPatientExtension(patient, answers['race'].valueCoding);
+  setPatientExtension(patient, answers['ethnicity'].valueCoding);
+
+  const sexualOrientationValueCoding = answers['sexual-orientation'].valueCoding;
+  if (sexualOrientationValueCoding) {
+    await medplum.upsertResource(getSexualOrientationObservation(patient, sexualOrientationValueCoding), {
+      code: '76690-7',
+      subject: getReferenceString(patient),
+    });
+  }
+
+  // Handle language preferences
+
+  const languagesSpoken = answers['languages-spoken'];
+  if (languagesSpoken?.valueCoding) {
+    addPatientLanguage(patient, languagesSpoken.valueCoding);
+  }
+  const preferredLanguage = answers['preferred-language'];
+  if (preferredLanguage?.valueCoding) {
+    addPatientLanguage(patient, preferredLanguage.valueCoding, true);
+  }
+
+  await medplum.updateResource(patient);
+}
+
+function addPatientLanguage(patient: Patient, valueCoding: Coding, preferred: boolean = false): void {
+  const patientCommunications = patient.communication || [];
+
+  let language = patientCommunications.find(
+    (communication) => communication.language.coding?.[0].code === valueCoding?.code
+  );
+
+  if (!language) {
+    language = {
+      language: {
+        coding: [valueCoding],
+      },
+    };
+    patientCommunications.push(language);
+  }
+
+  if (preferred) {
```

Comment:
@rahul1 I'm currently not ensuring there's only one preferred language set, LMK if this is a requirement